spectator-ext-jvm/src/main/java/com/netflix/spectator/jvm/Jmx.java

```diff
@@ -36,15 +35,51 @@ private Jmx() {
    * mbeans from the local jvm.
    */
   public static void registerStandardMXBeans(Registry registry) {
+    monitorClassLoadingMXBean(registry);
+    monitorThreadMXBean(registry);
+    monitorCompilationMXBean(registry);
+
     for (MemoryPoolMXBean mbean : ManagementFactory.getPlatformMXBeans(MemoryPoolMXBean.class)) {
       registry.register(new MemoryPoolMeter(registry, mbean));
     }
-
     for (BufferPoolMXBean mbean : ManagementFactory.getPlatformMXBeans(BufferPoolMXBean.class)) {
       registry.register(new BufferPoolMeter(registry, mbean));
     }
   }
 
+  private static void monitorClassLoadingMXBean(Registry registry) {
+    ClassLoadingMXBean classLoadingMXBean = ManagementFactory.getPlatformMXBean(ClassLoadingMXBean.class);
+    PolledMeter.using(registry)
+      .withName("jvm.classloading.classesLoaded")
+      .monitorMonotonicCounter(classLoadingMXBean, ClassLoadingMXBean::getTotalLoadedClassCount);
+    PolledMeter.using(registry)
+      .withName("jvm.classloading.classesUnloaded")
+      .monitorMonotonicCounter(classLoadingMXBean, ClassLoadingMXBean::getUnloadedClassCount);
+  }
+
+  private static void monitorThreadMXBean(Registry registry) {
+    ThreadMXBean threadMXBean = ManagementFactory.getPlatformMXBean(ThreadMXBean.class);
+    PolledMeter.using(registry)
+      .withName("jvm.thread.startedThreadCount")
+      .monitorMonotonicCounter(threadMXBean, ThreadMXBean::getTotalStartedThreadCount);
+    PolledMeter.using(registry)
+      .withName("jvm.thread.threadCount")
```

Comment:
This could also be implemented as:

```
name=jvm.thread.count,id=[daemon,non-daemon]
```

Where non-daemon is:

```
(ThreadMXBean::getThreadCount - ThreadMXBean::getDaemonThreadCount)
```

This would allow for the sum of the metric to be the total count of live threads and offers a natural group-by.

The downside is duplicate calls to the `ThreadMXBean::getThreadCount` method and possible timing issues related to collecting data, if they are polled separately.

Reply:
I had the same thought myself, and the separate polling was why I didn't. Is there a way of doing a single measurement/poll for those?

Reply:
You could do something like:

```java
    Gauge nonDaemonThreadCount = registry.gauge("jvm.thread.threadCount", "id", "non-daemon");
    Gauge daemonThreadCount = registry.gauge("jvm.thread.threadCount", "id", "daemon");
    PolledMeter.poll(registry, () -> {
      int threads = threadMXBean.getThreadCount();
      int daemonThreads = threadMXBean.getDaemonThreadCount();
      nonDaemonThreadCount.set(Math.max(0, threads - daemonThreads));
      daemonThreadCount.set(daemonThreads);
    });
```

In theory the values could change between the two calls to get the counts, but should be fairly rare and typically the rough number of threads is mainly what this is indicating.